src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,51 @@
+package task;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * * Represents a collection of tasks in the Duke application.
+ */
+public class TaskList implements Iterable<Task> {
+
+    private List<Task> tasks = new ArrayList<>();
+
+    public TaskList() {}
+
+    public void addTasks(Task task) {
```

Comment:
```suggestion
    public void addTask(Task task) {
```
Perhaps it will be better if this method name is in singular form since each time you are only adding one task?